File: 1755-defuse-the-bomb/1755-defuse-the-bomb.cpp
class Solution {
public:
    vector<int> decrypt(vector<int>& code, int k) {
        int l = code.size();

        vector<int>res(l,0);
        if(k==0){
            return res;

        }else if(k<0){
            for(int i=0; i<l; i++){
                int K =k;
                while(K!=0){
                    int idx = (i+K+l)%l;
                    res[i] = res[i] + code[idx];
                    K++;
                }
            }

        }else{
            for(int i=0; i<l; i++){
                int K =k;
                while(K!=0){
                   int idx = (i+K)%l;
                   res[i]+= code[idx];
                    K--;
                }
            }

        }
        return res;
       
    }
};

File: 1755-defuse-the-bomb/1755-defuse-the-bomb.java
class Solution {
    public int[] decrypt(int[] code, int k) {
        int n = code.length;
        int[] res = new int[n];

        if (k == 0) {
          
            return res;
        } else if (k < 0) {
      
            for (int i = 0; i < n; i++) {
                int K = k;
                while (K != 0) {
                    int idx = (i + K + n) % n;
                    res[i] += code[idx];
                    K++;
                }
            }
        } else {
          
            for (int i = 0; i < n; i++) {
                int K = k;
                while (K != 0) {
                    int idx = (i + K) % n;
                    res[i] += code[idx];
                    K--;
                }
            }
        }
        return res;
    }
}